syntax = "proto3";

package file_retrieval;

// TO-DO update the service definition
service FileRetrieval {
    rpc Index(IndexRequest) returns (IndexReply) {}
    rpc Search(SearchRequest) returns (SearchReply) {}
}

// TO-DO update the IndexRequest message to contain the local index or files to index
message IndexRequest {
    string message = 1;
}

// TO-DO update the IndexReply message to contain the return values of indexing
message IndexReply {
    string message = 1;
}

// TO-DO update the SearchRequest message to contain the term to be searched
message SearchRequest {
    string message = 1;
}

// TO-DO update the SearchReply message to contain the results of a single term search
message SearchReply {
    string message = 1;
}